corehq/apps/email/forms.py

```diff
@@ -0,0 +1,143 @@
+    )
+
+    from_email = forms.EmailField(
+        label=_("Sender's email"),
+        required=True,
+    )
+
+    use_this_gateway = forms.BooleanField(
+        label=_("Use Gateway?"),
+        required=False,
+        help_text=_("Select this option to use this email gateway for sending emails")
+    )
+
+    use_tracking_headers = forms.BooleanField(
+        label=_("Use Tracking Headers"),
+        required=False,
+        help_text=_("Applicable for Amazon's gateway. When selected, "
+                    "emails will be sent with tracking headers along with other headers.")
+    )
+
+    sns_secret = forms.CharField(
+        label=_("SNS endpoint secret"),
+        required=False,
+        help_text=_("Applicable only when gateway is Amazon's SES and "
+                    "tracking headers are enabled. This secret "
+                    "is used in the AWS SNS settings.")
+    )
+
+    class Meta:
+        model = EmailSettings
+        fields = [
+            'username',
+            'plaintext_password',
+            'server',
+            'port',
+            'from_email',
+            'use_this_gateway',
+            'use_tracking_headers',
+            'sns_secret'
+        ]
+
+    def __init__(self, *args, **kwargs):
+        if kwargs.get('instance'):
+            # `plaintext_password` is not a database field, and so
+            # super().__init__() will not update `initial` with it. We
+            # need to do that here.
+            #
+            # We use PASSWORD_PLACEHOLDER to avoid telling the user what
+            # the password is, but still indicating that it has been
+            # set. (The password is only changed if its value is not
+            # PASSWORD_PLACEHOLDER.)
+            password = kwargs['instance'].plaintext_password
+            if 'initial' in kwargs:
+                kwargs['initial'].update({
+                    'plaintext_password': PASSWORD_PLACEHOLDER if password else '',
+                })
+            else:
+                kwargs['initial'] = {
+                    'plaintext_password': PASSWORD_PLACEHOLDER if password else '',
+                }
+        super().__init__(*args, **kwargs)
+
+    @cached_property
+    def helper(self):
+        helper = hqcrispy.HQFormHelper()
+        helper.layout = crispy.Layout(
+            twbscrispy.PrependedText('use_this_gateway', ''),
+            crispy.Field('username'),
+            crispy.Field('plaintext_password'),
+            crispy.Field('server'),
+            crispy.Field('port'),
+            crispy.Field('from_email'),
+            twbscrispy.PrependedText('use_tracking_headers', ''),
+            crispy.Field('sns_secret'),
+            hqcrispy.FormActions(
+                twbscrispy.StrictButton(
+                    _("Save"),
+                    type="submit",
+                    css_class="btn btn-primary",
+                ),
+            )
+        )
+        return helper
+
+    def clean_port(self):
```

Comment:
you can provide a `min_value` and `max_value` to `IntegerField` and it'll do the validation for you:
https://docs.djangoproject.com/en/4.1/ref/forms/fields/#integerfield

Reply:
Changed